Library/Homebrew/language/haskell.rb

```diff
@@ -63,8 +67,15 @@ def install_cabal_package(*args)
         cabal_sandbox do
           cabal_install_tools(*options[:using]) if options[:using]
 
+          # if we have build flags, we have to pass them to cabal install to resolve the necessary
+          # dependencies, and call cabal configure afterwards to set the flags again for compile
+          flags = options[:flags] ? ["--flags='#{options[:flags].join(" ")}'"] : []
```

Comment:
I think this might be a bit more readable:
```
flags = ""

if options[:flags]
  flags = "--flags='#{options[:flags].join(" ")}'"
end

args_and_flags = [*args, flags]

..

cabal_install "--only-dependencies", args_and_flags

..

cabal_configure flags unless flags.empty?
```

And make `cabal_configure` take a single string argument called `flags` (as you're never building an array)

...

```